source/evp_override.c

```diff
@@ -706,6 +707,35 @@ const EVP_MD *EVP_sha512() {
     return &md;
 }
 
+EVP_MD *EVP_MD_fetch(OSSL_LIB_CTX *ctx, const char *algorithm, const char *properties) {
+    // Assume this method is always called without a library context
+    assert(ctx != NULL);
+
+    // Reuse the old initialization values to set value
+    EVP_MD *md_value = NULL;
+    if (strcmp(algorithm, "MD5") == 0) {
+        md_value = EVP_md5();
+    } else if (strcmp(algorithm, "SHA1") == 0) {
+        md_value = EVP_sha1();
+    } else if (strcmp(algorithm, "SHA224") == 0) {
+        md_value = EVP_sha224();
+    } else if (strcmp(algorithm, "SHA256") == 0) {
+        md_value = EVP_sha256();
+    } else if (strcmp(algorithm, "SHA384") == 0) {
+        md_value = EVP_sha256();
```

Comment:
Errr... that should be ECP_sha384(); I think...

Reply:
Fixed. Apparently our proofs don't actually care what EVP_MD they get :cry: